app/main.py

```diff
@@ -1,3 +1,111 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def get_dict_from_json() -> dict:
+    with open("app/config.json", "r") as data:
+        data = json.load(data)
+
+    customers = [Customer(
+        name=customer["name"],
+        product_cart=customer["product_cart"],
+        location=customer["location"],
+        money=customer["money"],
+        car=Car(brand=customer["car"]["brand"],
+                fuel_consumption=customer["car"]["fuel_consumption"], )
+    ) for customer in data["customers"]]
+
+    shops = [Shop(
+        name=shop["name"],
+        products=shop["products"],
+        location=shop["location"],
+    ) for shop in data["shops"]]
+
+    return {
+        "FUEL_PRICE": data["FUEL_PRICE"],
+        "customers": customers,
+        "shops": shops,
+    }
+
+
+def length_trip_calculation(x_point: list, y_point: list) -> float:
+    return math.hypot(y_point[0] - x_point[0], y_point[1] - x_point[1])
+
+
+def cost_trip_calculation(
+        length_trip: float,
+        fuel_consumption: float,
+        fuel_price: float,
+        cost_products: float,
+) -> float:
+    return round(
+        length_trip * fuel_consumption / 100
+        * fuel_price * 2 + cost_products, 2
+    )
+
+
+def shop_trip() -> None:
+    data = get_dict_from_json()
```

Comment:
u still have 50-row function
as a rule it should be 10-20 rows

for example: try separate "print part" from "logic part"